cuda_core/cuda/core/experimental/_memoryview.pyx

```diff
@@ -78,30 +79,72 @@ cdef class StridedMemoryView:
         bint readonly
         object exporting_obj
 
-    # If using dlpack, this is a strong reference to the result of
-    # obj.__dlpack__() so we can lazily create shape and strides from
-    # it later.  If using CAI, this is a reference to the source
-    # `__cuda_array_interface__` object.
-    cdef object metadata
-
-    # The tensor object if has obj has __dlpack__, otherwise must be NULL
-    cdef DLTensor *dl_tensor
-
-    # Memoized properties
-    cdef tuple _shape
-    cdef tuple _strides
-    cdef bint _strides_init  # Has the strides tuple been init'ed?
-    cdef object _dtype
-
-    def __init__(self, obj=None, stream_ptr=None):
+    cdef:
+        # If using dlpack, this is a strong reference to the result of
+        # obj.__dlpack__() so we can lazily create shape and strides from
+        # it later.  If using CAI, this is a reference to the source
+        # `__cuda_array_interface__` object.
+        object metadata
+
+        # The tensor object if has obj has __dlpack__, otherwise must be NULL
+        DLTensor *dl_tensor
+
+        # Memoized properties
+        tuple _shape
+        tuple _strides
+        # a `None` value for _strides has defined meaning in dlpack and
+        # the cuda array interface, meaning C order, contiguous.
+        #
+        # this flag helps prevent unnecessary recompuation of _strides
+        bint _strides_init
+        object _dtype
+
+    def __init__(self, obj: object = None, stream_ptr: int | None = None) -> None:
+        cdef str clsname = self.__class__.__name__
         if obj is not None:
             # populate self's attributes
             if check_has_dlpack(obj):
+                warnings.warn(
+                    f"Constructing a {clsname} directly from a DLPack-supporting object is deprecated; "
+                    "Use `StridedMemoryView.from_dlpack` or `StridedMemoryView.from_any_interface` instead."
+                )
```

Comment:
Can we make this a `FutureWarning` (https://docs.python.org/3/library/exceptions.html#FutureWarning) or whatever warning type is most appropriate for this deprecation?

Reply:
Done.